cmd/installer/pkg/install/preflight.go

```diff
@@ -258,7 +258,7 @@ func kubernetesVersionFromImageRef(ref string) (*compatibility.KubernetesVersion
 		return nil, fmt.Errorf("invalid image reference: %q", ref)
 	}
 
-	return compatibility.ParseKubernetesVersion(ref[idx+2:])
+	return compatibility.ParseKubernetesVersion(strings.Split(ref[idx+2:], "@")[0])
```

Comment:
this goes in the right direction, but we need a better fix from coding point of view

Reply:
I will try to fix it up